src/Hyperlink/index.tsx

```diff
@@ -1,13 +1,17 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import classNames from 'classnames';
+import {
+  type BsPrefixRefForwardingComponent as ComponentWithAsProp,
+  type BsPrefixProps,
+} from 'react-bootstrap/esm/helpers';
 import { Launch } from '../../icons';
 import Icon from '../Icon';
 
 export const HYPER_LINK_EXTERNAL_LINK_ALT_TEXT = 'in a new tab';
 export const HYPER_LINK_EXTERNAL_LINK_TITLE = 'Opens in a new tab';
 
-interface Props extends Omit<React.ComponentPropsWithRef<'a'>, 'href' | 'target'> {
+interface HyperlinkProps extends BsPrefixProps, Omit<React.ComponentPropsWithRef<'a'>, 'href' | 'target'> {
   /** specifies the URL */
   destination: string;
```

Comment:
When using `Hyperlink` with `Link` from React Router (or `GatsbyLink` on the Paragon docs site example), the `destination` prop will not suffice as it gets passed to an `href` prop.

Both `Link` and `GatsbyLink` utilize the standard hyperlink (i.e., `a`) when `href` prop is provided; to get it to work with internal routing, the `to` prop must be passed.

Because we use prop spreading of all attributes passed to the `Hyperlink` component, consumers _can_ pass `to` prop but would continue to need to pass `destination` as it's a required prop.

We will need to determine how consumers can pass `to` without needing to also pass an identical value to `destination` (i.e., `href`), e.g.:

```jsx
<Hyperlink
  as={GatsbyLink}
  // `destination` is still a required prop even though the `to` takes precedence.
  destination="/components/button"
  to="/components/button"
>
  Button
</Hyperlink>
```

Reply:
@bradenmacdonald Curious if you might have any thoughts on how to mitigate this concern around `to` vs. `href` vs. `destination`?

Reply:
Yeah, I was wondering about this.

I think one nice option is to change the Paragon `hyperlink` to accept `to` instead of `destination` and just treat `destination` as a deprecated alias for `to` ?

Otherwise to avoid the need to introspect `as` and determine what prop to pass based on some conditions, you could change the logic that currently converts `destination` to `href` and instead convert `destination` to _both_ `href` and `to`, and unconditionally pass both props to the inner `as` component. Though I think this may display a DOM warning in the console that `a` doesn't accept a `to` attribute. So maybe some conditional logic is actually needed.